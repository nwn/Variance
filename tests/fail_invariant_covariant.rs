mod common;

use common::*;

fn invariant_fail_covariant<'a>() {
    let _in: In<Lifetime<'a>> = In(
        Invariant::<Lifetime<'static>>::default(),
    );
}

fn main() {}
