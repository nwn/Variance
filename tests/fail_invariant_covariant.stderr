error[E0308]: mismatched types
 --> $DIR/fail_invariant_covariant.rs:7:9
  |
7 |         Invariant::<Lifetime<'static>>::default(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
  |
  = note: expected struct `type_variance::Invariant<common::Lifetime<'a>>`
             found struct `type_variance::Invariant<common::Lifetime<'static>>`
note: the lifetime `'a` as defined on the function body at 5:29...
 --> $DIR/fail_invariant_covariant.rs:5:29
  |
5 | fn invariant_fail_covariant<'a>() {
  |                             ^^
  = note: ...does not necessarily outlive the static lifetime
